### DISCUSSION TITLE
Inquiry on Automated Disk Scanning and Registration for Large-Scale Longhorn Deployment

### DISCUSSION THREAD

**Discussion by jangseon-ryu** [longhorn/longhorn#9797] [Q&A] [answered]
Hello Longhorn Community,

I am currently evaluating a transition to a Longhorn Engine v2 environment to support our extensive volume services, currently managed with a local provisioner across thousands of nodes. In this scenario, I am exploring options to automate the detection and registration of disks on each node.

#### Background:

- At present, my understanding is that disk registration in Longhorn requires manually adding each disk via the UI on a per-node basis.
- Given the scale of our infrastructure (thousands of nodes), this manual approach would entail significant operational resource investment.
- Automating this process would greatly improve efficiency and reduce resource overhead.

#### Question:

1. Does Longhorn currently support any features or methods for automatic disk scanning and registration on nodes?
2. Alternatively, is there an easier method or any scripts that could facilitate the disk registration process across multiple nodes?
3. If not, is there any roadmap or future plan to introduce such a capability?

--- 

Thank you for your time and any insights you might be able to share. This feature would be incredibly valuable for organizations managing large-scale Longhorn deployments.

Best regards,
Jason

**Comment by akashsemil** [comment, marked as answer]
Have you tried the following ?

https://longhorn.io/docs/1.7.2/nodes-and-volumes/nodes/default-disk-and-node-config/

**Reply by jangseon-ryu** [reply]
Thank you for your response. I'll test it out. :D

**Reply by jangseon-ryu** [reply]
@akashsemil 
With the guidance provided, I tested the automatic disk registration process, and it worked well—thank you for your support! 
However, I noticed that the Storage Reserved value for all automatically registered disks is set to 0.
```
    enginev2-disk1:
      Allow Scheduling:    true
      Disk Driver:         auto
      Disk Type:           block
      Eviction Requested:  false
      Path:                /dev/disk/by-id/scsi-3600508b1001c95d27f513bddea3acc4b
      Storage Reserved:    0  <--
      Tags:
```
_**_Does the **_automatic disk registration process not apply the storage-reserved-percentage-for-default-disk setting_** to determine the Storage Reserved value?_**_
```
storage-reserved-percentage-for-default-disk                      30
```

( cc @derekbit )
